src/file-uploader/index.tsx

```diff
@@ -0,0 +1,247 @@
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { create, tsx } from '@dojo/framework/core/vdom';
+import {
+	FileUploadInput,
+	FileUploadInputChildren,
+	FileUploadInputProperties,
+	ValidationInfo
+} from '../file-upload-input';
+import { Icon } from '../icon';
+import theme from '../middleware/theme';
+import bundle from './nls/FileUploader';
+
+import * as css from '../theme/default/file-uploader.m.css';
+import * as fileUploadInputCss from '../theme/default/file-upload-input.m.css';
+import * as fileUploadInputFixedCss from '../file-upload-input/styles/file-upload-input.m.css';
+
+export interface FileUploaderChildren {
+	label?: RenderResult;
+}
+
+export interface FileUploaderProperties extends FileUploadInputProperties {
+	/** Custom validator used to validate each file */
+	customValidator?: (file: File) => ValidationInfo | void;
+
+	/** The files to render in the widget (controlled scenario) */
+	files?: FileWithValidation[];
```

Comment:
Seems odd that the validation is being mixed into the files for all of these callbacks. Would have expected `onValue` etc and `files` to only take / pass `File[]` params.